src/core/connection.c

```diff
@@ -2884,6 +2889,19 @@ QuicConnProcessPeerTransportParameters(
         "Peer Transport Parameters Set");
     Connection->State.PeerTransportParameterValid = TRUE;
 
+    if (Connection->Stats.GreaseBitNegotiated == FALSE &&
+        (Connection->PeerTransportParams.Flags & QUIC_TP_FLAG_GREASE_QUIC_BIT) > 0) {
```

Comment:
I don't think we should have this extra condition:
```suggestion
    if (Connection->PeerTransportParams.Flags & QUIC_TP_FLAG_GREASE_QUIC_BIT) {
```

Reply:
And actually, shouldn't we add a condition for only if the local setting is enabled?

Reply:
And, let's move this to the end of the function too please.

Reply:
> And actually, shouldn't we add a condition for only if the local setting is enabled?

Yes! Looks like I forgot it, I'll add it.